lib/Optimizer/Transforms/ArgumentSynthesis.cpp

```diff
@@ -163,3 +163,12 @@ cudaq::opt::createArgumentSynthesisPass(ArrayRef<StringRef> funcNames,
   return std::make_unique<ArgumentSynthesisPass>(
       ArgumentSynthesisOptions{pairs});
 }
+
+std::unique_ptr<mlir::Pass> cudaq::opt::createArgumentSynthesisPass(
+    const std::vector<std::string> &funcNames,
```

Comment:
As discussed, we should see if we can't get rid of this overload and use the original function with the LLVM ADT signature.